util/stream.go

```diff
@@ -49,28 +70,39 @@ type MessageStream struct {
 	Outbound chan Message
 	// Channel on which to receive a shutdown command
 	Shutdown chan bool
+	// Worker to parse the message received from the connection
+	workers []*streamWorker
+	// Stream context to control the worker goroutines
+	cxt context.Context
+	// Function to shut down the parser goroutines
+	cancelFunc context.CancelFunc
```

Comment:
it's a bit unusual to use a `Context` object like this IMO. I'm curious about why you didn't use a simple channel?

Reply:
Sure, I would change to use the original `parserShutdown` channel.